modules/javafx.graphics/src/main/java/javafx/scene/CssStyleHelper.java

```diff
@@ -571,6 +571,140 @@ boolean pseudoClassStateChanged(PseudoClass pseudoClass) {
 
     }
 
+    // This method is a reduced version of transitionToState() method, it is added as a fix for JDK-8204568.
+    // Any modifications to the method transitionToStates() should be relatively applied here if needed.
+    // Currently this method is executed only by Labeled.fontProperty().set(), when it's font size
+    // is changed by LabeledText.
+    void recalculateRelativeSizeProperties(final Node node, Font fontForRelativeSizes) {
+
+        if (transitionStateInProgress) {
+            // If transitionToState() is being executed for the current control then all the css properties will get
+            // calculated there, and so we need to do anything here.
+            return;
+        }
+        if (cacheContainer == null) {
+            return;
+        }
+        final StyleMap styleMap = getStyleMap(node);
+        if (styleMap == null) {
+            return;
+        }
+        // if the style-map is empty, then we are only looking for inherited styles.
+        final boolean inheritOnly = styleMap.isEmpty();
+
+        final Set<PseudoClass>[] transitionStates = getTransitionStates(node);
+        CalculatedValue cachedFont = new CalculatedValue(fontForRelativeSizes, null, false);
+
+        final List<CssMetaData<? extends Styleable,  ?>> styleables = node.getCssMetaData();
+        final int numStyleables = styleables.size();
+
+        for (int n = 0; n < numStyleables; n++) {
+
+            final CssMetaData<Styleable,Object> cssMetaData =
```

Comment:
This may need the ` @SuppressWarnings("unchecked")` annotation, as done in the original method.

Reply:
Added back the annotation.